cinder/volume/drivers/vmware/fcd.py

```diff
@@ -492,7 +492,11 @@ def copy_image_to_volume(self, context, volume, image_service, image_id):
         profile_id = self._get_storage_profile_id(volume)
         if profile_id:
             self.volumeops.update_fcd_policy(fcd_loc, profile_id)
-
+        try:
+            LOG.warning("Extending volume %s to %s", volume.id, volume.size)
+            self.volumeops.extend_fcd(fcd_loc, volume['size'] * units.Ki)
+        except Exception as e:
+            LOG.error(e)
```

Comment:
Are you sure we should continue if we couldn't extend the volume to the proper size? Could have many reasons, only one of them being that the volume is already at that size.

Reply:
I think we can refine this in a follow up.